### ISSUE TITLE
IMAP support?

### ISSUE THREAD

**Issue by Mic92** [cherti/mailexporter#31]
For full end-to-end testing IMAP support would be useful and also is easier to integrate into existing infrastructure since the exporter does not need to run on the mailserver. Is this a feature you would accept as a PR? If so is there a preferred IMAP library?

**Comment by cherti** [comment]
Yes, I am willing to merge IMAP support. It is something that I have already though of myself, as it has many benefits, as it would also allow monitoring the sending capabilities, which mailexporter cannot really be used for currently.

As for preferred IMAP-libraries, I would prefer to keep the dependency footprint of mailexporter low, so IMAP libraries that do not require tons of dependencies themselves would be preferred, libraries only needing the standard libraries would therefore of course be perfect, if there is one.

Fair warning, due to current private time constraints iterations on a PR might take a bit, won't mean I have forgotten you, just that my capacities are currently rather limited. :slightly_smiling_face:

**Comment by Mic92** [comment]
There only seems to be one active maintained library anyway: https://github.com/emersion/go-imap The only third-party dependency seems golang.org/x/text/transform which could be counted as an extended stdlib.

**Comment by cherti** [comment]
Yes, that does look very promising indeed.
